### ISSUE TITLE
Automate versioning

### ISSUE THREAD

**Issue by Polymeta** [3Squared/ForgeUI#41]
Remove that manual step of bumping versions. 

With this repo containing multiple packages, do we want to potentially give all packages the same version and bump them together (ts-eslint monorepo does that)? Or keep them all individual, if that's even possible. 

Do we want to use conventional commits for version bumps then (and also validate the PR title etc to make sure they are in correct format)?

I assume we want to publish everytime we merge into main, right?

When doing this, should be also investigate deploying beta versions to npm for testing (or a separate registry)?

Just some thoughts/opinions needed around this, before we can put this automation in place.

**Comment by TomSmith27** [comment]
Yep really good comments, 

I'm not sure about the best way to do it in a monorepo.

i know this tool exists https://github.com/changesets/changesets

Which is supposed to help but haven't had much time to investigate

**Comment by Polymeta** [comment]
Oh yeah, that tool looks very neat, even comes with a bot to spread the information about which PRs will cause what updates to get pushed, quite cool I think.

I can look into that and see how it works, thanks for the pointer. Maybe this helps us with beta releases too somehow, if not we can make a dedicated manually triggered workflow for this potentially

**Comment by TomSmith27** [comment]
Awesome that would be amazing!
